tron.bat

```diff
@@ -774,19 +774,14 @@ if /i %VERBOSE%==yes call functions\log.bat "%CUR_DATE% %TIME% !  VERBOSE (-v) o
 
 
 :: Run a quick SMART check and notify if there are any drives with problems
-%WMIC% diskdrive get status > "%TEMP%\tron_smart_results.txt"
-setlocal enabledelayedexpansion
-for %%i in (Error,Degraded,Unknown,PredFail,Service,Stressed,NonRecover) do (
-	%FIND% /i "%%i" %TEMP%\tron_smart_results.txt >nul 2>&1
-	if !ERRORLEVEL!==0 (
-		call functions\log.bat "%CUR_DATE% %TIME% ^^^! WARNING: SMART check indicates at least one drive with '%%i' status"
-		call functions\log.bat "%CUR_DATE% %TIME%   SMART errors can mean a drive is close to failure"
-		call functions\log.bat "%CUR_DATE% %TIME%   Recommend you back the system up BEFORE running Tron."
-		color 0e
-		ENDLOCAL DISABLEDELAYEDEXPANSION && set WARNINGS_DETECTED=yes&& goto smart_check_complete
-	)
+set WARNING_LIST=(Error Degraded Unknown PredFail Service Stressed NonRecover)
+for /f %%i in ('%WMIC% diskdrive get status') do echo %%i|findstr /i "%WARNING_LIST:~1,-1%" && (
+	call functions\log.bat "%CUR_DATE% %TIME% ^^^! WARNING: SMART check indicates at least one drive with '%%i' status"
+	call functions\log.bat "%CUR_DATE% %TIME% SMART errors can mean a drive is close to failure"
+	call functions\log.bat "%CUR_DATE% %TIME% Recommend you back the system up BEFORE running Tron."
+	color 0e
+	set WARNINGS_DETECTED=yes
```

Comment:
P.S. I removed the `goto smart_check_complete` here because we want the output to indicate each error type if more than one error type exists in case of multiple drives. Example C: might be Stressed and D: might be Error.